app/javascript/mastodon/reducers/domain_lists.js

```diff
@@ -0,0 +1,23 @@
+import {
+  DOMAIN_BLOCKS_FETCH_SUCCESS,
+  DOMAIN_BLOCKS_EXPAND_SUCCESS,
+  DOMAIN_UNBLOCK_SUCCESS,
+} from '../actions/domain_blocks';
+import { Map as ImmutableMap, List as ImmutableList } from 'immutable';
+
+const initialState = ImmutableMap({
+  blocks: ImmutableMap(),
+});
+
+export default function userLists(state = initialState, action) {
+  switch(action.type) {
+  case DOMAIN_BLOCKS_FETCH_SUCCESS:
+    return state.setIn(['blocks', 'items'], ImmutableList(action.domains)).setIn(['blocks', 'next'], action.next);
```

Comment:
What about using OrderedSet instead of List?

Reply:
Hm… good question, I initially copied how `user_lists` worked, but `OrderedSet` seems fine. I'll try using it.